core/include/assert.h

```diff
@@ -41,6 +41,19 @@ extern const char assert_crash_message[];
  *
  * The purpose of this macro is to help programmers find bugs in their
  * programs.
+ *
+ * The output if an assertion fails and NDEBUG is set is usually something like
```

Comment:
There are some false assumptions (in the original wording as well):

1. There is a logical error: the output only is provided if `NDEBUG` is not defined
2. There is IMHO a difference between an enabled (or set) macro and a defined macro. To demonstrate lets take the antonyms "disabled" and "undefined": a disabled macro can be defined, but its value is set to 0, while an undefined macro just does not exist.

Except for the wording though I agree with the simpler sentencing. Will fix.